### PR TITLE
Processing thread fixes

### PR DESCRIPTION
**Pull request by sergey-chernikov**
Mainly for Dealer usage, not used widely in the other code.

### CODE REVIEW

File: CommonLib/ProcessingThread.h [hunk 1]
```diff
@@ -79,41 +89,62 @@ class ProcessingThread
...
+                     }
+                     packets.emplace_back(std::move(packet));
+                  }
+                  delayedPackets_.erase(itDelayed);
+               }
             }
 
-            if (pendingPackets_.empty()) {
-               pendingPacketsEvent_.ResetEvent();
+            while (!pendingPackets_.empty()) {
```
Comment (goatpig) [inline CommonLib/ProcessingThread.h:124]: Why the busy loop? This kind of approach suggests other threads may append pendingPackets_ while this thread is consuming its content, but a lock is acquired on pendingPacketLock_ at the begining of this scope.

Comment (sergey-chernikov) [reply CommonLib/ProcessingThread.h:124]: Not exactly. This loop just suggests batch processing if multiple packets were added before lock was acquired.

Comment (goatpig) [reply CommonLib/ProcessingThread.h:124]: Why does this require a while loop though? What does it provide that this doesnt:
packets.insert(packets.end(), pendingPackets_.begin(), pendingPackets.end())

Comment (sergey-chernikov) [reply CommonLib/ProcessingThread.h:124]: `pendingPackets_.clear();`
but anyway this is another way of doing the same thing

**Review by goatpig — APPROVED** [review]
tACK

**Pull request merged**